File: CS/CustomReportClass.Module.Win/CustomXAFReport.cs
using DevExpress.ExpressApp.Reports;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using DevExpress.XtraReports.Serialization;

namespace CustomReportClass.Module {
    [DevExpress.XtraReports.RootClass]
    public class CustomXafReport : XafReport {
        protected string additionalInfo;
        public string AdditionalInfo {
            get { return additionalInfo; }
            set { additionalInfo = value; }
        }
        protected override void SerializeProperties(XRSerializer serializer) {
            base.SerializeProperties(serializer);
            serializer.SerializeString("AdditionalInfo", AdditionalInfo);
        }
        protected override void DeserializeProperties(XRSerializer serializer) {
            base.DeserializeProperties(serializer);
            AdditionalInfo = serializer.DeserializeString("AdditionalInfo", string.Empty);
        }
    }
    [System.ComponentModel.DisplayName("Custom Report")]
    [NavigationItem("Reports")]
    public class CustomReportData : ReportData {
        public CustomReportData(Session session) : base(session) { }
        protected override XafReport CreateReport() {
            return new CustomXafReport();
        }
        public string CustomProperty { get; set; }
    }
}

File: VB/CustomReportClass.Module.Win/CustomXAFReport.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports DevExpress.XtraReports.Serialization

Namespace CustomReportClass.Module
	Public Class CustomXafReport
		Inherits XafReport
		Protected additionalInfo_Renamed As String
		Public Property AdditionalInfo() As String
			Get
				Return additionalInfo_Renamed
			End Get
			Set(ByVal value As String)
				additionalInfo_Renamed = value
			End Set
		End Property
		Protected Overrides Sub SerializeProperties(ByVal serializer As XRSerializer)
			MyBase.SerializeProperties(serializer)
			serializer.SerializeString("AdditionalInfo", AdditionalInfo)
		End Sub
		Protected Overrides Sub DeserializeProperties(ByVal serializer As XRSerializer)
			MyBase.DeserializeProperties(serializer)
			AdditionalInfo = serializer.DeserializeString("AdditionalInfo", String.Empty)
		End Sub
	End Class
	<System.ComponentModel.DisplayName("Custom Report"), NavigationItem("Reports")> _
	Public Class CustomReportData
		Inherits ReportData
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Protected Overrides Function CreateReport() As XafReport
			Return New CustomXafReport()
		End Function
		Private privateCustomProperty As String
		Public Property CustomProperty() As String
			Get
				Return privateCustomProperty
			End Get
			Set(ByVal value As String)
				privateCustomProperty = value
			End Set
		End Property
	End Class
End Namespace
